include/psen_scan_v2/reply_msg_from_scanner.h

```diff
@@ -99,16 +112,12 @@ inline ReplyMsgFromScanner ReplyMsgFromScanner::fromRawData(const RawScannerData
 {
   ReplyMsgFromScanner msg{ 0, 0 };
 
-  // Alternatives for transformation:
-  // typedef boost::iostreams::basic_array_source<char> Device;
-  // boost::iostreams::stream<Device> stream((char*)&data, sizeof(DataReply::MemoryFormat));
-
-  std::istringstream stream(std::string((char*)&data, REPLY_MSG_FROM_SCANNER_SIZE));
+  std::istringstream is(std::string((char*)&data, REPLY_MSG_FROM_SCANNER_SIZE));
 
-  stream.read((char*)&msg.crc_, sizeof(ReplyMsgFromScanner::crc_));
-  stream.read((char*)&msg.reserved_, sizeof(ReplyMsgFromScanner::reserved_));
-  stream.read((char*)&msg.opcode_, sizeof(ReplyMsgFromScanner::opcode_));
-  stream.read((char*)&msg.res_code_, sizeof(ReplyMsgFromScanner::res_code_));
+  read<uint32_t>(is, msg.crc_);
```

Comment:
You can skip the template type here (and in the following lines). The type is automatically determined by the compiler.